### DISCUSSION TITLE
How can you use Insiders with Netlify?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2484] [Q&A] [answered]
This was already discussed with @squidfunk through Email where he suggested to me to make a discussion for this.

I essentially search for a way to use Material for MkDocs Insiders with Netlify. I currently use Netlify to preview Pull request changes and similar to many others do I get the dependencies using a `requirements.txt` file on the repository.
Unfortunately can you not use Env variables in the txt file, so I need to find another possible solution to work with.

If anyone here has any ideas on how you could implement this would I be thankful for it.

**Comment by squidfunk** [comment]
From my research, you could probably use a custom build command and point it to a shell script you add to your repository, which will `pip install ...` the requirements file and then Insiders:
https://docs.netlify.com/configure-builds/get-started/

Environment variables should also be possible to set:
https://docs.netlify.com/configure-builds/environment-variables/

However, I'm not an in with Netlify deployments.

**Comment by JulioV** [comment]
I added my token as a Github Actions secret, use `mike deploy` to push to the `gh_pages` branch, then in Netlify, I deploy whatever is on that branch (without a build command) on every push.

This is my [GA workflow config yaml](https://github.com/carissalow/rapids/blob/master/.github/workflows/docs.yaml)

These are my build settings in Netlify

```
Build settings

Repository
    github.com/carissalow/rapids
Base directory
    Not set
Build command
    Not set
Publish directory
    Not set
Deploy log visibility
    Logs are public
Builds
    Active
```

And my deploy contexts

```
Deploy contexts

Deploy contexts are a way to tell Netlify how to build your site. They give you the flexibility to configure your site’s builds depending on the context they are going to be deployed to.

Production branch
    gh-pages
Deploy previews
    Any pull request against your production branch / branch deploy branches
Branch deploys
    Deploy only the production branch
```

**Reply by Andre601** [reply]
The problem I see with this is, that it won't work with Pull requests.

**Reply by JulioV** [reply]
You can always `mike deploy` to a new alias and that alias can be the PR hash or some other ID. You could then add another GA trigger to delete that alias when you close a PR (or just keep old builds or delete them manually).

**Reply by Andre601** [reply]
Not everyone uses Mike.

**Reply by JulioV** [reply]
sure, good luck!

**Comment by squidfunk** [comment, marked as answer]
After a quick discussion on Twitter, @StarfallProjects wrote a guide on how to deploy Insiders on Netlify:
https://www.starfallprojects.co.uk/posts/deploy-mkdocs-insiders-netlify/

**Reply by StarfallProjects** [reply]
Thanks for sharing that! Just want to note I'm always open to feedback, and if someone comes up with a better way, I'd love to know - I'm still not wild about GitHub tokens visible in the Netlify GUI.

**Reply by squidfunk** [reply]
They're visible? I thought that Netlify somehow allows to declare variables as sensitive, see:
https://docs.netlify.com/configure-builds/environment-variables/#sensitive-variable-policy

**Reply by StarfallProjects** [reply]
As I understand it, that controls whether the autobuilds have access / who can trigger a build? But not whether the variable is visible in the GUI to team members?

By visible I mean, this:

![image](https://user-images.githubusercontent.com/10284570/119645307-ceea8a00-be15-11eb-8f28-933782108753.png)

I may be misunderstanding

**Reply by tahamaddam** [reply]
Above link throws 404. Here, I believe is the correct link https://www.starfallprojects.co.uk/projects/deploy-host-docs/deploy-mkdocs-material-netlify/
